src/webots/gui/WbMainWindow.cpp

```diff
@@ -1321,13 +1321,22 @@ bool WbMainWindow::proposeToSaveWorld(bool reloading) {
 bool WbMainWindow::loadWorld(const QString &fileName, bool reloading) {
   if (!proposeToSaveWorld(reloading))
     return true;
+  if (!WbApplication::instance()->isValidWorldFile(fileName))
+    return false;  // invalid filename, abort without affecting the current simulation
   mSimulationView->cancelSupervisorMovieRecording();
   logActiveControllersTermination();
-  return WbApplication::instance()->loadWorld(fileName, reloading);
+  WbLog::setConsoleLogsPostponed(true);
+  const bool success = WbApplication::instance()->loadWorld(fileName, reloading);
+  if (!success) {
+    WbLog::setConsoleLogsPostponed(false);
+    WbLog::showPostponedPopUpMessages();
+  }
+  // else will be reset in updateAfterWorldLoading()
```

Comment:
Do you mean the console will be cleared in updateAfterWorldLoading()?

Reply:
No, I mean that the `WbLog::setConsoleLogsPostponed` flag will be reset and the postponed messages will be forwarded in the updateAfterWorldLoading.

Reply:
In that case, I would make it clearer in the comment:
```suggestion
  // else will the WbLog::setConsoleLogsPostponed() flag be reset in updateAfterWorldLoading()
```

Reply:
I improved the comment:
``// else console messages will be forwarded after world load in restorePerspective()``